components/camel-workday/src/main/java/org/apache/camel/component/workday/auth/AuthClientForIntegration.java

```
@@ -0,0 +1,111 @@
+import org.apache.commons.httpclient.methods.PostMethod;
+
+import java.io.IOException;
+import java.util.Base64;
+
+public class AuthClientForIntegration implements AutheticationClient {
+
+    private static final String GRANT_TYPE = "grant_type";
+
+    private static final String REFRESH_TOKEN = "refresh_token";
+
+    private static final String CONTENT_TYPE_HEADER = "Content-Type";
+
+    private static final String AUTHORIZATION_HEADER = "Authorization";
+
+    private static final String ACCESS_TOKEN = "access_token";
+
+    private static final String CONTENT_TYPE = "application/x-www-form-urlencoded";
+
+    public static final String BASE_TOKEN_ENDPOINT = "https://%s/ccx/oauth2/%s/token";
+
+    private WorkdayConfiguration workdayConfiguration;
+
+    public AuthClientForIntegration(WorkdayConfiguration workdayConfiguration) {
+        this.workdayConfiguration = workdayConfiguration;
+    }
+
+    @Override
+    public void configure(HttpClient httpClient, HttpMethodBase method) {
+
+        String bearerToken = getBearerToken(httpClient);
+        method.addRequestHeader(AUTHORIZATION_HEADER, "Bearer " + bearerToken);
+
+    }
+
+    protected String getBearerToken(HttpClient httpClient) {
+
+        String tokenUrl = String.format(BASE_TOKEN_ENDPOINT,
+                workdayConfiguration.getHost(),
+                workdayConfiguration.getTenant());
+
+        PostMethod postMethod = createPostMethod(tokenUrl);
+
+        try {
+            int statusCode = httpClient.executeMethod(postMethod);
+
+            if (statusCode != HttpStatus.SC_OK) {
+                throw new IllegalStateException("Got the invalid http status value '" + postMethod.getStatusLine() + "' as the result of the Token Request '" + tokenUrl + "'");
+            }
+
+            String response = postMethod.getResponseBodyAsString();
+            return parseResponse(response);
+
+        } catch (IOException e) {
+            e.printStackTrace();
```

I'd throw an `RuntimeException` instead of printing the stack trace